sdks/java/core/src/main/java/org/apache/beam/sdk/schemas/io/SchemaCapableIOProvider.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.sdk.schemas.io;
+
+import org.apache.beam.sdk.schemas.Schema;
+import org.apache.beam.sdk.values.Row;
+
+/** Provider to create {@link SchemaIO}. */
+public interface SchemaCapableIOProvider {
```

Comment:
Sorry I should've brought this up before. We should add some disclaimers here and in SchemaIO (and the exceptions should at least get the annotations):

```suggestion
/** 
 * Provider to create {@link SchemaIO} instances for use in Beam SQL and other SDKs.
 *
 * <p><b>Internal only:</b> This interface is actively being worked on and it will likely change as we provide
 * implementations for more standard Beam IOs. We provide no backwards compatibility guarantees and
 * it should not be implemented outside of the Beam repository.
 */
@Internal
@Experimental(Kind.SCHEMAS)
public interface SchemaCapableIOProvider {
```

I think it's very likely we'll make some changes to these interfaces (e.g. @tysonjh and @kennknowles have a good [suggestion](https://docs.google.com/document/d/1ic3P8EVGHIydHQ-VMDKbN9kEdwm7sBXMo80VrhwksvI/edit?disco=AAAAGn4u7-4) in the doc), so we don't want anyone other than us trying to implement them yet. Let's do this to make that clear.